sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientImpl.java

```diff
@@ -617,15 +617,20 @@ static String unpackAndValidateId(String keyId) {
 
     static LocalKeyCryptographyClient initializeCryptoClient(JsonWebKey jsonWebKey,
                                                              CryptographyClientImpl implClient) {
-        if (jsonWebKey.getKeyType().equals(RSA) || jsonWebKey.getKeyType().equals(RSA_HSM)) {
-            return new RsaKeyCryptographyClient(jsonWebKey, implClient);
-        } else if (jsonWebKey.getKeyType().equals(EC) || jsonWebKey.getKeyType().equals(EC_HSM)) {
-            return new EcKeyCryptographyClient(jsonWebKey, implClient);
-        } else if (jsonWebKey.getKeyType().equals(OCT) || jsonWebKey.getKeyType().equals(OCT_HSM)) {
-            return new AesKeyCryptographyClient(jsonWebKey, implClient);
-        } else {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException(String.format(
-                "The JSON Web Key type: %s is not supported.", jsonWebKey.getKeyType().toString())));
+        try {
+            if (jsonWebKey.getKeyType().equals(RSA) || jsonWebKey.getKeyType().equals(RSA_HSM)) {
+                return new RsaKeyCryptographyClient(jsonWebKey, implClient);
+            } else if (jsonWebKey.getKeyType().equals(EC) || jsonWebKey.getKeyType().equals(EC_HSM)) {
+                return new EcKeyCryptographyClient(jsonWebKey, implClient);
+            } else if (jsonWebKey.getKeyType().equals(OCT) || jsonWebKey.getKeyType().equals(OCT_HSM)) {
+                return new AesKeyCryptographyClient(jsonWebKey, implClient);
+            } else {
+                throw LOGGER.logExceptionAsError(new IllegalArgumentException(String.format(
+                    "The JSON Web Key type: %s is not supported.", jsonWebKey.getKeyType().toString())));
+            }
+        } catch (RuntimeException e) {
+            throw LOGGER.logExceptionAsError(
+                new RuntimeException("Could not initialize local cryptography client.", e));
```

Comment:
Change this try/catch to include everything but the `else` block above, we are catching the exception we just threw and wrapping it in another layer of exceptions.